src/main/java/com/google/flourbot/api/Bot.java

```diff
@@ -69,12 +72,28 @@ public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecu
         break;
       case "MESSAGE":
         // Sends request to execution module
-        String email = event.at("/message/sender/email").asText();
         String message = event.at("/message/text").asText();
         String threadId = event.at("/message/thread/name").asText();
-        replyText = macroExecutionModule.execute(email, message, threadId);
+
+        String[] words = message.split(" "); 
+
+        if (words[1].equalsIgnoreCase("initiate")) {
+            String creator = event.at("/message/sender/email").asText();
+            roomToCreator.put(event.at("/space/name").asText(), creator);
+            replyText = "Your macros have now been initiated for this room. All users in this room can use your macros.";
+        }
+        else {
+            if (roomToCreator.containsKey(event.at("/space/name").asText())) {
+                String email = roomToCreator.get(event.at("/space/name").asText());
+                replyText = macroExecutionModule.execute(email, message, threadId);
+            }
+            else {
+                replyText = "You do not have access to this macro.";
+            }
+        }
         break;
       case "REMOVED_FROM_SPACE":
+        roomToCreator.remove(event.at("/space/name").asText());
```

Comment:
Hmmm... if we go with the room_macro as the key, the cleanup will be harder. Maybe the proper structure is Map<String, Map<String, String>> - first the room as the key, then the macro name.